### DISCUSSION TITLE
Sharing my config for an SKR-2 with a TFT35 E3 V3.0

### DISCUSSION THREAD

**Discussion by levidurham** [teeminus/NoTouchScreenFirmware#62] [General]
[board_pins]
aliases:
    # EXP1 header
    EXP1_1=PC5, EXP1_3=PB1, EXP1_5=PE10, EXP1_7=PE12, EXP1_9=<GND>,
    EXP1_2=PB0, EXP1_4=PE9, EXP1_6=PE11, EXP1_8=PE13, EXP1_10=<5V>,
    # EXP2 header
    EXP2_1=PA6, EXP2_3=PE7, EXP2_5=PB2, EXP2_7=PC4,   EXP2_9=<GND>,
    EXP2_2=PA5, EXP2_4=PA4, EXP2_6=PA7, EXP2_8=<RST>, EXP2_10=<NC>

[display]
lcd_type: emulated_st7920
spi_software_miso_pin: EXP1_6
spi_software_mosi_pin: EXP1_3
spi_software_sclk_pin: EXP1_5
en_pin: EXP1_4
encoder_pins: ^EXP2_5, ^EXP2_3
click_pin: ^!EXP1_2

**Comment by teeminus** [comment]
Thank you very much, I added the config to the wiki.

**Reply by levidurham** [reply]
I realized I had a regular, non-E3 TFT35 V3.0 laying around and tested it with the same config. All working.

**Reply by teeminus** [reply]
Yeah, the Klipper display config only depends on the mainboard, not on the display.

**Reply by rjsantos47** [reply]
Hopefully this will work for mine. I will test it tomorrow. Is this for the new chip SKR2??

**Reply by rjsantos47** [reply]
This what i keep getting. 

Option 'exp1_1' is not valid in section 'board_pins'

**Reply by teeminus** [reply]
You need to paste the complete content of the first post to your Klipper config.

**Reply by rjsantos47** [reply]
So i need to add all the files that has .h on it and my config folder? Im not great with all of this and still having hard time to navigate on this github thing. Hope you could help me out. Thank you

**Reply by teeminus** [reply]
No. In the first post of this discussion, there is a working display config. Paste the content of the post to your Klipper config and restart Klipper.

I have no idea what you mean by the .h files.

**Reply by rjsantos47** [reply]
I added this to my printer.cfg 

board_pins]
aliases:
# EXP1 header
EXP1_1=PC5, EXP1_3=PB1, EXP1_5=PE10, EXP1_7=PE12, EXP1_9=,
EXP1_2=PB0, EXP1_4=PE9, EXP1_6=PE11, EXP1_8=PE13, EXP1_10=<5V>,
# EXP2 header
EXP2_1=PA6, EXP2_3=PE7, EXP2_5=PB2, EXP2_7=PC4, EXP2_9=,
EXP2_2=PA5, EXP2_4=PA4, EXP2_6=PA7, EXP2_8=, EXP2_10=

[display]
lcd_type: emulated_st7920
spi_software_miso_pin: EXP1_6
spi_software_mosi_pin: EXP1_3
spi_software_sclk_pin: EXP1_5
en_pin: EXP1_4
encoder_pins: ^EXP2_5, ^EXP2_3
click_pin: ^!EXP1_2



And im getting error on

Option 'exp1_1' is not valid in section 'board_pins'

I really apologize if I don't get what your saying on the first post. My new to all this things. I did went to the wiki and copy the file from there. If that's what your saying. Thank you again

**Reply by teeminus** [reply]
Can you upload your "printer.cfg" file here, please?

**Reply by rjsantos47** [reply]
[printer.txt](https://github.com/teeminus/NoTouchScreenFirmware/files/8913037/printer.txt)

thank you again.

**Reply by teeminus** [reply]
Why are the `[board_pins]` and `[display]` sections commented out in your config?

Pasting the following to your printer config should work:
```
[board_pins]
aliases:
EXP1_1=PC5, EXP1_3=PB1, EXP1_5=PE10, EXP1_7=PE12, EXP1_9=,
EXP1_2=PB0, EXP1_4=PE9, EXP1_6=PE11, EXP1_8=PE13, EXP1_10=<5V>,
EXP2_1=PA6, EXP2_3=PE7, EXP2_5=PB2, EXP2_7=PC4, EXP2_9=,
EXP2_2=PA5, EXP2_4=PA4, EXP2_6=PA7, EXP2_8=, EXP2_10=

[display]
lcd_type: emulated_st7920
spi_software_miso_pin: EXP1_6
spi_software_mosi_pin: EXP1_3
spi_software_sclk_pin: EXP1_5
en_pin: EXP1_4
encoder_pins: ^EXP2_5, ^EXP2_3
click_pin: ^!EXP1_2
```

**Reply by rjsantos47** [reply]
I put the # coz when i uncommented it and do a save and restart. I get the error Option 'exp1_1' is not valid in section 'board_pins'

**Comment by RolandLarocque** [comment]
[printer.txt](https://github.com/teeminus/NoTouchScreenFirmware/files/9277290/printer.txt)
Attached is my printer.cfg file.

I am also running skr2 with tft35v3.  

The display is showing "ST7920 Emulator Ready"

I also get the error :
Option 'exp1_1' is not valid in section 'board_pins'

Once the underlying issue is corrected, use the "RESTART"
command to reload the config and restart the host software.
Printer is halted

**Reply by teeminus** [reply]
Someone else has previously reported a similar error. I have no idea what causes it.

As a workaround you could try removing the `board_pins` section and insert the pin names directly to the display config section. Eg. `spi_software_miso_pin: PE11`
